tests/test_parser_requirements.py

```diff
@@ -31,87 +31,106 @@ def test_simple(self) -> None:
-        self.assertFalse(parser.has_warnings())
+        components = parser.get_components()
+
+        self.assertFalse(parser.has_warnings(), f'{parser.get_warnings()}')
+        self.assertEqual(1, len(components), f'{components}')
 
     def test_example_1(self) -> None:
         with open(os.path.join(os.path.dirname(__file__),
                                'fixtures/requirements-example-1.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-        self.assertTrue(3, parser.component_count())
-        self.assertFalse(parser.has_warnings())
+        components = parser.get_components()
+
+        self.assertFalse(parser.has_warnings(), f'{parser.get_warnings()}')
+        self.assertEqual(3, len(components), f'{components}')
 
     def test_example_with_comments(self) -> None:
         with open(os.path.join(os.path.dirname(__file__),
                                'fixtures/requirements-with-comments.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-        self.assertTrue(5, parser.component_count())
-        self.assertFalse(parser.has_warnings())
+        components = parser.get_components()
+
+        self.assertFalse(parser.has_warnings(), f'{parser.get_warnings()}')
+        self.assertEqual(5, len(components), f'{components}')
 
     def test_example_multiline_with_comments(self) -> None:
         with open(os.path.join(os.path.dirname(__file__),
                                'fixtures/requirements-multilines-with-comments.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-        self.assertTrue(5, parser.component_count())
-        self.assertFalse(parser.has_warnings())
+        components = parser.get_components()
+
+        self.assertFalse(parser.has_warnings(), f'{parser.get_warnings()}')
+        self.assertEqual(5, len(components), f'{components}')
 
     def test_example_local_packages(self) -> None:
         with open(os.path.join(os.path.dirname(__file__),
                                'fixtures/requirements-local-and-remote-packages.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-        self.assertTrue(6, parser.component_count())
-        self.assertFalse(parser.has_warnings())
+        components = parser.get_components()
+
+        self.assertFalse(parser.has_warnings(), f'{parser.get_warnings()}')
+        self.assertEqual(6, len(components), f'{components}')
 
     def test_example_local_and_nested_packages(self) -> None:
         # RequirementsFileParser can parse nested requirements files,
         # but RequirementsParser cannot.
         parser = RequirementsFileParser(
             requirements_file='fixtures/requirements-local-and-remote-packages.txt'
         )
-        self.assertTrue(7, parser.component_count())
```

Comment:
this assertion was wrong. it used to mean `isEqual()`.
i gnot this fixed, and now the tests fail, because the implemetation does not find `7` pacckages (as correctly expected),
but `6`

Reply:
issue was introduced via #327
@mostafa FYI.

Reply:
@mostafa gould you branch out from `bugfix/tests-are-broken-wrong`
then have the issues fixed
and pullrequest into `bugfix/tests-are-broken-wrong` ?

Reply:
Addressed.